js/ui/grid_core/ui.grid_core.focus.js

```diff
@@ -774,13 +772,21 @@ module.exports = {
                             rowIndex += that.getController("data").getRowIndexOffset();
                             that.getController("keyboardNavigation").setFocusedCellPosition(rowIndex, columnIndex);
 
+                            const dataSource = that.component.getController("data")._dataSource;
+                            const operationTypes = dataSource && dataSource.operationTypes();
                             if(operationTypes && !operationTypes.paging) {
                                 that.resizeCompleted.remove(that._scrollToFocusOnResize);
                                 that.resizeCompleted.add(that._scrollToFocusOnResize);
                             }
                         }
                     }
                 },
+                _getFocusedRowElement: function() {
+                    const focusedRowKey = this.option("focusedRowKey");
+                    const rowIndex = this._dataController.getRowIndexByKey(focusedRowKey);
```

Comment:
This is wrong. The getRowIndexByKey return visible row index. But focusedRowKey for virtual scrolling return global row index

Reply:
Here I need visible index to find focusedRow element in the visible part